datashader/core.py

```diff
@@ -264,7 +264,11 @@ def raster(self,
         w = int(np.ceil(self.plot_width * width_ratio))
         h = int(np.ceil(self.plot_height * height_ratio))
 
-        data = resample_2d(source.values[0].astype(np.float32),
+        rmin, cmin = get_indices(xmin, ymin, source.x.values, source.y.values, res)
+        rmax, cmax = get_indices(xmax, ymax, source.x.values, source.y.values, res)
```

Comment:
Can you clarify what you mean by "this" fixing it?  Do you need to push a change to this PR before it can be merged, or do you mean the PR is already correct?